test/integration/test_datatype_upload_dos.py

```diff
@@ -0,0 +1,161 @@
+        <backend id="files2" type="disk" weight="1">
+            <files_dir path="${temp_directory}/database/files2"/>
+            <extra_dir type="temp" path="${temp_directory}/database/tmp2"/>
+            <extra_dir type="job_work" path="${temp_directory}/database/job_working_directory2"/>
+        </backend>
+    </backends>
+</object_store>
+""")
+IRODS__OBJECT_STORE_CONFIG = string.Template("""
+<object_store type="distributed">
+    <backends>
+        <backend id="files1" type="disk" weight="1">
+            <files_dir path="${temp_directory}/database/files1"/>
+            <extra_dir type="temp" path="${temp_directory}/database/tmp1"/>
+            <extra_dir type="job_work" path="${temp_directory}/database/job_working_directory1"/>
+        </backend>
+        <backend id="files2" type="irods" weight="1">
+            <auth username="${username}" password="${password}"/>
+            <resource name="${resource}"/>
+            <zone name="${zone}"/>
+            <connection host="${host}" port="${port}" timeout="${timeout}"/>
+            <cache path="${temp_directory}/object_store_cache" size="1000"/>
+            <extra_dir type="job_work" path="${temp_directory}/job_working_directory_irods"/>
+            <extra_dir type="temp" path="${temp_directory}/tmp_irods"/>
+        </backend>
+    </backends>
+</object_store>
+""")
+
+
+def start_irods(container_name):
+    irods_start_args = [
+        'docker',
+        'run',
+        '-p',
+        '1247:1247',
+        '-d',
+        '--name',
+        container_name,
+        'kxk302/irods-server:0.1']
+    subprocess.check_call(irods_start_args)
+
+    # Sleep so integration test's iRODS instance is up and running
+    time.sleep(20)
+
+
+def stop_irods(container_name):
+    subprocess.check_call(['docker', 'rm', '-f', container_name])
+
+
+class UploadTestDosDiskAndDiskTestCase(UploadTestDatatypeDataTestCase):
+
+    @classmethod
+    def handle_galaxy_config_kwds(cls, config):
+        temp_directory = cls._test_driver.mkdtemp()
+        print('temp_directory:', temp_directory)
+        cls.object_stores_parent = temp_directory
+        config_path = os.path.join(temp_directory, "object_store_conf.xml")
+        # This doesn't quite work yet, fails with extra_files_path
+        # config["metadata_strategy"] = "extended"
+        config["outpus_to_working_dir"] = True
+        config["retry_metadata_internally"] = False
+        config["object_store_store_by"] = "uuid"
+        with open(config_path, "w") as f:
+            f.write(
+                DISK_OBJECT_STORE_CONFIG.safe_substitute(
+                    {
+                        "temp_directory": temp_directory,
+                    }
+                )
+            )
+        config["object_store_config_file"] = config_path
+
+
+class UploadTestDosIrodsAndDiskTestCase(UploadTestDatatypeDataTestCase):
+
+    @classmethod
+    def handle_galaxy_config_kwds(cls, config):
+        temp_directory = cls._test_driver.mkdtemp()
```

Comment:
I noticed that in test_datatype_upload_irods.py there's a class definition like this:
```
class UploadTestDatatypeDataTestCase(UploadTestDatatypeDataTestCase):
```
Can you change this to `IrodsUploadTestDatatypeDataTestCase` and then inherit from this class for UploadTestDosIrodsAndDiskTestCase ? I think with that you can drop `setUpClass ` and `tearDownClass` from `UploadTestDosIrodsAndDiskTestCase`

I also think you can skip the duplicated logic in handle_galaxy_config_kwds and do something like 

```
    def handle_galaxy_config_kwds(cls, config):
        UploadTestDatatypeDataTestCase.handle_galaxy_config_kwds(config)
        temp_directory = cls.object_stores_parent
        with open(config_path, "w") as f:
            f.write(
                IRODS__OBJECT_STORE_CONFIG.safe_substitute(
                    {
                        "temp_directory": temp_directory,
                        "host": IRODS_OBJECT_STORE_HOST,
                        "port": IRODS_OBJECT_STORE_PORT,
                        "timeout": IRODS_OBJECT_STORE_TIMEOUT,
                        "username": IRODS_OBJECT_STORE_USERNAME,
                        "password": IRODS_OBJECT_STORE_PASSWORD,
                        "resource": IRODS_OBJECT_STORE_RESOURCE,
                        "zone": IRODS_OBJECT_STORE_ZONE
                    }
                )
            )
        config["object_store_config_file"] = config_path
```

Reply:
I extended the class as you suggested (good idea). However, I don't think I an re-use the code in handle_galaxy_config_kwds as they are operating on 2 different configs (OBJECT_STORE_CONFIG in irods test file, and IRODS_OBJECT_STORE_CONFIG is dos test).

Reply:
You could make it work. The example I posted above should take care of using `IRODS__OBJECT_STORE_CONFIG`, it was just missing `config_path = os.path.join(temp_directory, "object_store_conf.xml")`. It's not super-important, but I think it would help maintenance of these tests.

Reply:
A nicer approach could be creating a `object_store_config` class attribute that hold the expanded object store config, so the base class can just write this to disk and set the appropriate config value.

Reply:
Ok. I aded t class variables to the super class to store the config and config path. Will push the code now for review.